### DISCUSSION TITLE
How to apply a continuous-changing parameter in a Poisson problem using PartionedArrays and Gridap

### DISCUSSION THREAD

**Discussion by ZiwenH** [PartitionedArrays/PartitionedArrays.jl#139] [Q&A]
Hey guys,

I'd like to solve a problem with the bilinear form like following:
a(u,v) = ∫((vec)⋅∇(u)⋅∇(v))dΩ
where _vec_ is of Vector{Float64} type. But in parallel mode, it will lead to the error:
**ERROR: Scalar indexing on PVector is not allowed for performance reasons.**
There is no similar case in the tutorial, does anybody have any idea or experience how to treat the parameter?

Best regards

**Comment by fverdugo** [comment]
Hi @ZiwenH,

This question is more related with GridapDistributed than with PartitionedArrays.

In any case, you first need to wrap your vector into a CellField object

```julia
f = CellField(partition(vec),Ω)
a(u,v) = ∫(f⋅∇(u)⋅∇(v))dΩ
```
If this does not solve the issue, then ask in https://github.com/gridap/GridapDistributed.jl
